### ISSUE TITLE
nested nodegroups not available in ext_pillar

### ISSUE THREAD

**Issue by westry9** [saltstack/salt#45402]
### Description of Issue/Question
I'm using ext_pillar to get nodegroup data. I see any simple nodegroups but not any that contain other nodegroups. 

### Setup

**master.d**
```
ext_pillar:
  - nodegroups:
      pillar_name: 'nodegroups'
```

**nodegroups. this example comes straight from Saltstack docs**
```
nodegroups:
  group2: 'G@os:CentOS and foo.domain.com'
  group3: 'G@os:CentOS and N@group1'
```

group2 works well. I have trouble with group3.

### Steps to Reproduce Issue

- Changed master.d data listed above
- Restarted salt-master
- Refreshed pillar
- Check pillar.items to see if group2 exists where it should. It does.
- Check pillar.items to see if group3 exists where it should. It does **not**.

However, I do see group3 if I check from the command line as such:
`$ sudo salt -N group3 test.ping`

### Versions Report

```
$ sudo salt --versions-report
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: 1.6.0
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.24.6
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.24.2
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.11.6.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core

```

**Comment by Ch3LL** [comment]
could possibly be related to https://github.com/saltstack/salt/issues/45334 or https://github.com/saltstack/salt/issues/42735

can you give the workaround's in those issues a try?

**Comment by westry9** [comment]
#45334

This one does not look relevant to me. The workaround in that one was to avoid using YAML list lines. I'm not using them. However, I have tried the opposite just for fun and it still fails.

#42735

This one also appears to be a YAML listing issue, which I'm not doing.

**Comment by Paulo-Nunes** [comment]
`group3: 'G@os:CentOS and N@group1'`

Did you perhaps mean N@group2 ? I don't see group1 defined

**Comment by westry9** [comment]
Yes, sorry. The docs have all three groups. I took group1 out because it wasn't adding anything to the discussion.

```
nodegroups:
  group2: 'G@os:CentOS and foo.domain.com'
  group3: 'G@os:CentOS and N@group2'
```

**Comment by rjc** [comment]
This issue should not have been closed - the problem is still there.
Is anyone working on/looking into this?

**Comment by Ch3LL** [comment]
@rjc no one is currently working on this, please feel free to submit a fix.

**Comment by rjc** [comment]
Does this mean the issue has been resolved or is it `WONTFIX`?

**Comment by Ch3LL** [comment]
just needs to be re-opened

**Comment by remichristiaan** [comment]
Probably a related case is when you use PCRE (E@) in the nodegroup definition. 

E.g.:

```
nodegroups:
   group_a: 'E@^([a-z]{3,}.(sub_a|sub_b).([0-9]{2,})$'

```
It's possible to target the nodegroup: 

`salt -N group_a test.ping`

However the data under the ext_pillar entry for the nodegroup's file_tree is not available/loaded. 

`salt -N group_a pillar.items`  and `salt '*.sub_a.*' pillar.items` do not have the node's corresponding file_tree data. 

File_tree data for individual hosts shows up normally. 

This is tested with the 19.3 and 3000 salt versions.

**Comment by dwoz** [comment]
Cannot reproduce on 3006.x, 3007.x, 3008.x, or master. The fix landed as c9f834fba12 ("account for failed expansions of nodegroups", May 2018) and is present on every supported branch. nodegroup_comp recurses correctly through N@ references and the nodegroups ext_pillar reports membership in nested groups. Repro: drove salt.pillar.nodegroups.ext_pillar with the reporter's exact config against the real CkMinions compound logic; got {'nodegroups': ['group2', 'group3']} on all four branches. Please reopen with a repro on a currently supported version if you still see this.
